src/client/store/pages/assessmentSection/hooks/index.ts

```diff
@@ -43,6 +45,35 @@ export const useTableData = (props: { table: Table }): TableData => {
   return tableDataWithODP as TableData
 }
 
+export const useIsSectionDataEmpty = (tableSections: TableSection[]) => {
+  const countryIso = useCountryIso()
+  const { data, dataLoaded } = useAppSelector((state) => state.pages.assessmentSection)
+
+  const [sectionDataEmpty, setSectionDataEmpty] = useState(false)
+  const sectionTableNames = tableSections.flatMap((ts) => ts.tables.flatMap((t) => t.props.name))
+  const allTablesEmpty = useMemo(
+    () =>
+      sectionTableNames.every((tableName) =>
+        TableDatas.isTableDataEmpty({
+          data,
+          tableName,
+          countryIso,
+        })
+      ),
+    [countryIso, data, sectionTableNames]
+  )
```

Comment:
If we dont use `useMemo` here, we end in an infinite loop `setState <-> useEffect`